shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/UTxO.hs

```diff
@@ -223,10 +224,10 @@ makeWitnessesFromScriptKeys txbodyHash hashKeyMap scriptHashes =
    in makeWitnessesVKey txbodyHash (Map.elems witKeys)
 
 -- | Determine the total balance contained in the UTxO.
-balance :: Crypto crypto => UTxO crypto -> Coin
-balance (UTxO utxo) = foldr addCoins 0 utxo
+balance :: UTxO crypto -> Coin
+balance (UTxO utxo) = fromIntegral $ Map.foldl' addCoins 0 utxo
```

Comment:
Note that we use `Word64` to sum them and only convert to `Coin` at the end. Is this fine or do we expect an overflow?

Reply:
a property of the system is that there are never more than 45 * 10 ^ 15 coins in this map, so 2 ^ 64 is more that enough. that said, if there ever were a bug, this could obfuscate the problem.

Reply:
> a property of the system is that there are never more than 45 * 10 ^ 15 coins in this map, so 2 ^ 64 is more that enough.

I see.

> that said, if there ever were a bug, this could obfuscate the problem.

So what do you prefer?

Reply:
FYI, when summing using `Coin` instead of `Word64`:
```
time                 36.17 ms   (35.04 ms .. 37.35 ms)
                     0.998 R²   (0.996 R² .. 0.999 R²)
mean                 36.61 ms   (36.15 ms .. 37.22 ms)
std dev              1.112 ms   (812.8 μs .. 1.637 ms)
```

Reply:
I don't have a strong preference, I'm just making sure we all have the same understanding. Thank you for helping out with the performance @mrBliss !